### PR TITLE
Support data only array members for ``BindC`` structs

### PR DESCRIPTION
**Pull request by czgdp1807**
https://github.com/lcompilers/lpython/issues/1333#issuecomment-1359550823

Closes https://github.com/lcompilers/lpython/issues/1333

### CODE REVIEW

**Review by certik — APPROVED** [review]
I think this looks god now. I would add some tests.

**Comment by czgdp1807** [comment]
> I would add some tests.

`integration_tests/structs_14.py` already has some and I will add more there. Do you have any other use case which you want me to cover?

File: integration_tests/structs_14.py [hunk 1]
```diff
@@ -1,32 +1,72 @@
...
+    buffer4: f64[32]
 
 @ccallable
 @dataclass
 class buffer_struct_clink:
     buffer: i8[32]
+    buffer1: i32[32]
+    buffer2: f32[32]
+    buffer3: c32[32]
+    buffer4: f64[32]
```
Comment (certik) [inline integration_tests/structs_14.py:20]: Let's do also i16 and i64, those are commonly used. Let's do c64 too, might as well. Then everything is covered.

**Review by certik — APPROVED** [review]
I think that this is good. 

We can later see if there is a way to simplify some of these things, but for now this seems good enough and fixes the immediate problem.

**Pull request merged**